Frameworks/Foundation/NSOrderedSet.mm

```diff
@@ -14,311 +14,403 @@
 */
 - (instancetype)init {
-    UNIMPLEMENTED();
-    return StubReturn();
+    return [super init];
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
 - (BOOL)containsObject:(id)object {
-    UNIMPLEMENTED();
-    return StubReturn();
+    return mapContainer.find(object) != mapContainer.end();
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
 - (void)enumerateObjectsAtIndexes:(NSIndexSet*)indexSet
                           options:(NSEnumerationOptions)opts
                        usingBlock:(void (^)(id, NSUInteger, BOOL*))block {
-    UNIMPLEMENTED();
+    if (block == nil) {
+        [NSException raise:NSInvalidArgumentException format:@"block can not be nil."];
+    }
+
+    [indexSet enumerateIndexesWithOptions:opts
+                               usingBlock:^(NSUInteger idx, BOOL* stop) {
+                                   block([self objectAtIndex:idx], idx, stop);
+                               }];
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
 - (void)enumerateObjectsUsingBlock:(void (^)(id, NSUInteger, BOOL*))block {
-    UNIMPLEMENTED();
+    return [[self array] enumerateObjectsUsingBlock:block];
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
 - (void)enumerateObjectsWithOptions:(NSEnumerationOptions)opts usingBlock:(void (^)(id, NSUInteger, BOOL*))block {
-    UNIMPLEMENTED();
+    return [[self array] enumerateObjectsWithOptions:opts usingBlock:block];
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
 - (id)objectAtIndex:(NSUInteger)index {
-    UNIMPLEMENTED();
-    return StubReturn();
+    if (index >= [self count]) {
+        [NSException raise:NSRangeException format:@"index is greater than the orderedSet size %u", [self count]];
+    }
+    return setContainer[index];
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
+*/
+- (id)firstObject {
+    return [self objectAtIndex:0];
+}
+
+/**
+ @Status Interoperable
+*/
+- (id)lastObject {
+    return [self objectAtIndex:([self count] - 1)];
+}
+
+/**
+ @Status Interoperable
 */
 - (id)objectAtIndexedSubscript:(NSUInteger)index {
-    UNIMPLEMENTED();
-    return StubReturn();
+    return [self objectAtIndex:index];
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
 - (NSArray*)objectsAtIndexes:(NSIndexSet*)indexes {
-    UNIMPLEMENTED();
-    return StubReturn();
+    if (indexes == nil) {
+        [NSException raise:NSInvalidArgumentException format:@"indexes can not be nil."];
+    }
+
+    NSMutableArray* ret = [NSMutableArray array];
+    [indexes enumerateIndexesUsingBlock:^(NSUInteger idx, BOOL* stop) {
+        [ret addObject:[self objectAtIndex:idx]];
+    }];
+
+    return [[ret copy] autorelease];
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
 - (NSUInteger)indexOfObject:(id)object {
-    UNIMPLEMENTED();
-    return StubReturn();
+    if (![self containsObject:object]) {
+        return NSNotFound;
+    }
+    return mapContainer[object] - setContainer.begin();
```

Comment:
std::distance may be a better choice as pointer math can be hard to get right / hard to understand to the casual observer.